api.go

```diff
@@ -992,3 +1142,91 @@ func (api *API) State() string {
 func (api *API) Version() string {
 	return strings.TrimPrefix(Version, "v")
 }
+
+// API validation constants.
+const (
+	apiClusterMessage int = iota
+	apiCreateField
+	apiCreateFrame
+	apiCreateIndex
+	apiCreateInputDefinition
+	apiDeleteField
+	apiDeleteFrame
+	apiDeleteIndex
+	apiDeleteInputDefinition
+	apiDeleteView
+	apiExportCSV
+	apiFields
+	apiFragmentBlockData
+	apiFragmentBlocks
+	apiFrameAttrDiff
+	//apiHosts // not implemented
+	apiImport
+	apiImportValue
+	apiIndex
+	apiIndexAttrDiff
+	apiInputDefinition
+	//apiLocalID // not implemented
+	//apiLongQueryTime // not implemented
+	apiMarshalFragment
+	//apiMaxInverseSlices // not implemented
+	//apiMaxSlices // not implemented
+	apiModifyFrameTimeQuantum
+	apiModifyIndexTimeQuantum
+	apiQuery
+	apiRecalculateCaches
+	apiRemoveNode
+	apiResizeAbort
+	apiRestoreFrame
+	//apiSchema // not implemented
+	apiSetCoordinator
+	apiSliceNodes
+	//apiState // not implemented
+	//apiStatsWithTags // not implemented
+	apiUnmarshalFragment
+	//apiVersion // not implemented
+	apiViews
+	apiWriteInput
+)
+
+var functionCommon = []int{
+	apiClusterMessage,
+	apiMarshalFragment,
+	apiSetCoordinator,
+}
+
+var functionResizing = []int{
+	apiResizeAbort,
+}
+
+var functionNormal = []int{
```

Comment:
did you consider a `map[int]struct{}` for these? Perf difference is probably negligible, but it might be a bit more obvious what it's for at a glance.

Reply:
I don't quite understand this one. What does the map represent?